src/common/transformations/src/transformations/common_optimizations/sequence_fusion.cpp

```diff
@@ -0,0 +1,324 @@
+}
+
+bool check_WRB(const shared_ptr<RNNCellBase>& cell_1, const shared_ptr<RNNCellBase>& cell_2) {
+    int64_t idx_W = 2, idx_R = 3, idx_B = 4;
+
+    auto lstm_cell_1 = dynamic_pointer_cast<LSTMCell>(cell_1);
+    auto lstm_cell_2 = dynamic_pointer_cast<LSTMCell>(cell_2);
+    // 2nd input is Cell State
+    if (lstm_cell_1 && lstm_cell_2) {
+        ++idx_B;
+        ++idx_R;
+        ++idx_W;
+    }
+
+    auto lW = cell_1->input_value(idx_W).get_node_shared_ptr();
+    auto lR = cell_1->input_value(idx_R).get_node_shared_ptr();
+    auto lB = cell_1->input_value(idx_B).get_node_shared_ptr();
+    auto rW = cell_2->input_value(idx_W).get_node_shared_ptr();
+    auto rR = cell_2->input_value(idx_R).get_node_shared_ptr();
+    auto rB = cell_2->input_value(idx_B).get_node_shared_ptr();
+    bool is_equal = (lW.get() == rW.get() || is_equal_consts(lW, rW));
+    is_equal = is_equal && (lR.get() == rR.get() || is_equal_consts(lR, rR));
+    is_equal = is_equal && (lB.get() == rB.get() || is_equal_consts(lB, rB));
+    return is_equal;
+}
+
+bool is_equal_cells(const shared_ptr<RNNCellBase>& cell_1, const shared_ptr<RNNCellBase>& cell_2) {
+    bool is_equal = true;
+    auto gru_cell_1 = dynamic_pointer_cast<GRUCell>(cell_1);
+    auto gru_cell_2 = dynamic_pointer_cast<GRUCell>(cell_2);
+    if (gru_cell_1 && gru_cell_2) {
+        is_equal = gru_cell_1->get_linear_before_reset() == gru_cell_2->get_linear_before_reset();
+    }
+    is_equal = is_equal && cell_1->get_type_name() == cell_2->get_type_name() &&
+               cell_1->get_hidden_size() == cell_2->get_hidden_size() &&
+               cell_1->get_activations() == cell_2->get_activations() &&
+               cell_1->get_activations_alpha() == cell_2->get_activations_alpha() &&
+               cell_1->get_activations_beta() == cell_2->get_activations_beta() &&
+               cell_1->get_clip() == cell_2->get_clip() && check_WRB(cell_1, cell_2);
+    return is_equal;
+}
+
+shared_ptr<RNNCellBase> find_cell_chain(ov::pass::NodeRegistry& cp_from,
+                                        ov::pass::NodeRegistry& cp_to,
+                                        const shared_ptr<RNNCellBase>& current_cell,
+                                        ov::OutputVector& x_to_concat,
+                                        ov::OutputVector& attention_to_concat,
+                                        map<int, ov::Output<ov::Node>>& h_outputs_to_redirect,
+                                        int& cells_cnt,
+                                        const shared_ptr<ov::Node>& axis_1) {
+    cells_cnt = 1;
+
+    shared_ptr<RNNCellBase> current = current_cell;
+    while (true) {
+        cp_from.add(current);
+        // check the source node of HiddenState input
+        auto prev = current->input_value(1).get_node_shared_ptr();
+        if (auto prev_cell = dynamic_pointer_cast<RNNCellBase>(prev)) {
+            if (!is_equal_cells(prev_cell, current)) {
+                break;
+            }
```

Comment:
(Just a corner case)
What if the "not equal" cell will be at the end of 20 equal cells? 
The code from `else` block won't be reached. Is it ok?

Reply:
not ok :) thanks
fixed